gen/vsup_zimbra

```diff
@@ -110,23 +121,23 @@ foreach my $rData ($data->getChildElements) {
 				$users->{$uco}->{'TYPE'} = $relationType;
 				$users->{$uco}->{'COS'} = 'e69e98ba-e16b-4bdd-bc2c-0ae5e697337d';
 			} else {
-				die "User: '$uAttributes{$A_LOGIN}' was both GENERIC type and $users->{$uco}->{'TYPE'} type.";
+				die "User: '$login' was both GENERIC type and $users->{$uco}->{'TYPE'} type.";
 			}
 		} elsif ($relationType eq "MML") {
 			# CAN'T OVERLAP WITH ANOTHER
 			unless ($users->{$uco}->{'TYPE'}) {
 				$users->{$uco}->{'TYPE'} = $relationType;
 				$users->{$uco}->{'COS'} = '867a2875-136f-40be-aab8-751658db0300';
 			} else {
-				die "User: '$uAttributes{$A_LOGIN}' was both MAILMAN-LIST type and $users->{$uco}->{'TYPE'} type.";
+				die "User: '$login' was both MAILMAN-LIST type and $users->{$uco}->{'TYPE'} type.";
 			}
 		}
 
 		# get users name
-		my $firstName = ($uAttributes{$A_ARTISTIC_FIRST_NAME} || ($uAttributes{$A_FIRST_NAME} || ''));
-		my $lastName = ($uAttributes{$A_ARTISTIC_LAST_NAME} || ($uAttributes{$A_LAST_NAME} || ''));
+		my $firstName = ($artisticFirstName || ($uFirstName || ''));
+		my $lastName = ($artisticLastName || ($uLastName || ''));
```

Comment:
It would be nice, if this could be simplified, so we wouldn't need new ``$uFirstName`` or ``$uLastName`` variables. Maybe like this (for firstName):
```perl
my $firstName = $data->getUserAttributeValue(member => $memberId, attrName => $A_FIRST_NAME);
$firstName = ($artisticFirstName || ($firstName || ''));
```

I'm not sure about later ``$uDisplayName`` with ``$uDisplayName || undef;``. I think its meant to handle cases, when display name is defined but empty from Perun for following condition. If we update later ``unless`` condition, we can also use simple ``$displayName``, like:

```perl
my $displayName = $data->getUserAttributeValue(member => $memberId, attrName => $A_ZIMBRA_DISPLAY_NAME);
$displayName = $displayName || '';
unless(defined $displayName and length $displayName) {
    # construct display name from partial params
}
```
Then we could be sure, that at the end, all variables are defined:
```perl
$users->{$uco}->{"FIRST_NAME"} = $firstName;
$users->{$uco}->{"LAST_NAME"} = $lastName;
$users->{$uco}->{"DISPLAY_NAME"} = $displayName;
```

Reply:
Fixed